### PR TITLE
va-card: add prop to display a gray background

### PR DESCRIPTION
**Pull request by jamigibbs**
## Chromatic

https://2403-va-card-gray-prop--60f9b557105290003b387cd5.chromatic.com/?path=/docs/components-va-card--default

## Description
This PR adds a `background` prop to va-card that applies a gray background. It also refactors some of the style and classes to simplify the component.

Figma: https://www.figma.com/file/afurtw4iqQe6y4gXfNfkkk/VADS-Component-Library?type=design&node-id=2795-113&mode=design&t=iGMBkLHNOAVT7XSa-0

Closes https://github.com/department-of-veterans-affairs/vets-design-system-documentation/issues/2403

## QA Checklist
- [x] Component maintains 1:1 parity with design mocks
- [x] Text is consistent with what's been provided in the mocks
- [x] Component behaves as expected across breakpoints
- [ ] Accessibility expert has signed off on code changes (if applicable. If not applicable provide reason why)
- [ ] Designer has signed off on changes (if applicable. If not applicable provide reason why)
- [ ] Tab order and focus state work as expected
- [ ] Changes have been tested against screen readers (if applicable. If not applicable provide reason why)
- [x] New components are covered by e2e tests; updates to existing components are covered by existing test suite
- [ ] Changes have been tested in vets-website using Verdaccio (if applicable. If not applicable provide reason why)

## Screenshots

Screenshot 2024-01-18 at 12 55 03 PM

## Acceptance criteria
- [ ] QA checklist has been completed
- [ ] Screenshots have been attached that cover desktop and mobile screens

## Definition of done
- [ ] Documentation has been updated, if applicable
- [ ] A link has been provided to the originating GitHub issue (or connected to it via ZenHub)

### CODE REVIEW

File: packages/web-components/src/components/va-card/va-card.tsx [hunk 1]
```diff
@@ -23,15 +22,14 @@ export class VaCard {
...
-
-    const classes = classnames('va-card',  {'show-shadow': showShadow});
+  /**
+   * If `true`, the card will have a gray background.
+   */
+  @Prop() background?: boolean = false;
 
+  render() {
     return (
-      <Host class={classes}>
```
Comment (jamigibbs) [inline packages/web-components/src/components/va-card/va-card.tsx:34]: I think these classes were added originally to apply style to the host element but it's overcomplicating things. Style can be applied to the host using the `:host` selector and attributes like this instead:

```
:host([show-shadow]:not([show-shadow='false']))
```

**Review by micahchiang — APPROVED** [review]
:rocket:

**Comment by danbrady** [comment]
That is one sassy lookin' gray!

Because we added this background as separate, new prop, it's now possible to have a gray background with a drop shadow. We don't currently have that variation in Figma. Should we, @humancompanion-usds? Or should we set this up to be one prop that takes a string (i.e. "border", "shadow", "background"? I realize doing that might impact existing uses. (My mistake, I should have thought of this before making the changes.)

**Comment by jamigibbs** [comment]
> Because we added this background as separate, new prop, it's now possible to have a gray background with a drop shadow. We don't currently have that variation in Figma. Should we, @humancompanion-usds? Or should we set this up to be one prop that takes a string (i.e. "border", "shadow", "background"? I realize doing that might impact existing uses. (My mistake, I should have thought of this before making the changes.)

@danbrady  Maybe some useful info is that we don't currently have any instances of drop shadow being used at all. It was added for the `va-notification` component which is currently deprecated and will be removed at some point.

https://design.va.gov/storybook/?path=/docs/components-va-notification--default

**Comment by jamigibbs** [comment]
@danbrady Another thought is that we could add logic to the component so that drop-shadow can only be used with a white background. That seems to be what the component guidance mandates already:

> Use [this variation](https://design.va.gov/components/card#3-with-drop-shadow) for emphasis, for example at the top of a page or section, or as the primary card in a series of cards. **When using in a series, use with the default white variation. Do not combine this variation with the default gray variation.**

https://design.va.gov/components/card#variations

**Comment by danbrady** [comment]
> @danbrady Maybe some useful info is that we don't currently have any instances of drop shadow being used at all. It was added for the `va-notification` component which is currently deprecated and will be removed at some point.
> 
> https://design.va.gov/storybook/?path=/docs/components-va-notification--default

Hmm... I see @humancompanion-usds recently added the example below that shows a drop shadow version in the Claim Status Tool just [a few days ago](https://github.com/department-of-veterans-affairs/vets-design-system-documentation/commit/efe80bea48bcb8e269544bf16abf26180297d5c3).
 
image

**Comment by danbrady** [comment]
> @danbrady Another thought is that we could add logic to the component so that drop-shadow can only be used with a white background. That seems to be what the component guidance mandates already:
> 
> > Use [this variation](https://design.va.gov/components/card#3-with-drop-shadow) for emphasis, for example at the top of a page or section, or as the primary card in a series of cards. When using in a series, use with the default white variation. **Do not combine this variation with the default gray variation.**
> 
> https://design.va.gov/components/card#variations

I like that idea!

**Review by danbrady — APPROVED** [review]
LGTM!

**Pull request merged**